x/ccv/provider/keeper/keeper.go

```diff
@@ -282,10 +282,12 @@ func (k Keeper) SetConsumerChain(ctx sdk.Context, channelID string) error {
 
 	// the CCV channel is established:
 	// - set channel mappings
-	k.SetChainToChannel(ctx, tmClient.ChainId, channelID)
-	k.SetChannelToChain(ctx, channelID, tmClient.ChainId)
+	k.SetChainToChannel(ctx, chainID, channelID)
+	k.SetChannelToChain(ctx, channelID, chainID)
 	// - set current block height for the consumer chain initialization
-	k.SetInitChainHeight(ctx, tmClient.ChainId, uint64(ctx.BlockHeight()))
+	k.SetInitChainHeight(ctx, chainID, uint64(ctx.BlockHeight()))
+	// - remove init timeout timestamp
+	k.DeleteInitTimeoutTimestamp(ctx, chainID)
```

Comment:
Would be helpful to note in the godocs of these functions where it is called.

e.g. OnChanOpenConfirm

Since this is a complex application it would be easier for reviewers to quickly validate correctness, rather than having to find the caller themselves